experiment/build/docker_images.py

```diff
@@ -28,28 +28,74 @@ def _instantiate_image_obj(name_template, obj_template, fuzzer, benchmark):
     name = _substitute(name_template, fuzzer, benchmark)
     obj = obj_template.copy()
     for key in obj:
-        if key in ('build_arg', 'depends_on'):
-            obj[key] = [
-                _substitute(item, fuzzer, benchmark) for item in obj[key]
-            ]
+        if key in ('build_arg', 'depends_on', 'env_vars'):
+            obj[key] = [_substitute(it, fuzzer, benchmark) for it in obj[key]]
         else:
             obj[key] = _substitute(obj[key], fuzzer, benchmark)
     return name, obj
 
 
-def _get_image_type_templates():
+def _instantiate_string_gcb(image_types):
+    all_templates = {}
+    for name in image_types:
+        image = image_types[name]
+        if 'depends_on' in image.keys():
+            image['depends_on'] = [it.lstrip('.') for it in image['depends_on']]
+        all_templates[name.lstrip('.')] = image
+    return all_templates
+
+
+def _get_image_type_templates(oss_fuzz, skip_base):
     """Loads the image types config that contains "templates" describing how to
     build them and their dependencies."""
-    return yaml_utils.read('docker/image_types.yaml')
+    all_templates = yaml_utils.read('docker/image_types.yaml')
+    templates = {}
+    for name, image in all_templates.items():
+        if 'base' in name or 'dispatcher' in name:
+            if not skip_base:
+                templates[name] = [image]
+            continue
+        if oss_fuzz:
+            if 'oss-fuzz' in name:
+                templates[name] = image
+                continue
+        elif not oss_fuzz and not 'oss-fuzz' in name:
+            templates[name] = image
+    return templates
+
+
+def get_images_to_build_gcb(coverage=False):
```

Comment:
wondering, why do we have a separate gcb specific function for this? i see we're skipping dispatcher and base images. we never want to build these on gcb? 

(even if that's the case, would it make sense to filter them in the gcb spec generation script, and keep this module generic?)

Reply:
GCB needs debugging. I'll put as next CL.